src/biogeophys/CanopyStateType.F90

```diff
@@ -581,6 +581,7 @@ subroutine Restart(this, bounds, ncid, flag)
        call restartvar(ncid=ncid, flag=flag, varname='vegwp', xtype=ncd_double,  &
             dim1name='pft', dim2name='vegwcs', switchdim=.true., &
             long_name='vegetation water matric potential', units='mm', &
+            scale_by_thickness=.false., &
             interpinic_flag='interp', readvar=readvar, data=this%vegwp_patch) 
 
        call restartvar(ncid=ncid, flag=flag, varname='VEGWPLN', xtype=ncd_double,  &
```

Comment:
@billsacks 
I do not understand why I get the following error when I **add** `scale_by_thickness=.false., &` to the next two calls and get the same error when I **remove** `scale_by_thickness=.false., &` from the calls of "skipped" variables in subgridRestMod.F90 below. This is the error:
`There is no matching specific subroutine for this generic subroutine call.   [RESTARTVAR]`

As is the code runs, but it's inconsistent in that I am not including `scale_by_thickness=.false., &` in the next two calls.

Reply:
I think I understand now. The two calls that give the error when I add the argument `scale_by_thickness=...` do not include the argument `switchdim=...` so the compiler assumes I'm calling a version of `restartvar` that does not include `switchdim`. And `scale_by_thickness` is not recognized in that version of restartvar.

I see two options:
- Leave the two calls unchanged since init_interp skips these variables anyway
- Bring the `scale_by_thickness` code to this other version of restartvar because these variables, although skipped, have units of `mm` which I'm pretty sure makes them extensive properties
@billsacks pls let me know your preference.

Reply:
I often struggle with these different i/o subroutines that have long and subtly different argument lists. I think the ideal thing would be to make `restartvar_{DIMS}d_{TYPE}` only support 1d, not 2d, and force all 2-d restart calls to go through `restartvar_2d_{TYPE}_bounds`. My gut says that is probably too big of a rabbit hole to go down right now, though it would sure help long-term. I guess you could try and see what happens: change `!DIMS 1,2` to just `!DIMS 1` above that subroutine and see how many compilation errors you hit – temporarily removing or commenting out calls that give compilation errors as you go, to see how many there are total. If you hit more than just a few, stop: it isn't worth it right now. But if it's just a few, it might be worth doing this cleanup, and I could work with you on that.

But given my gut feeling that that might be too much of a rabbit hole, I think the next best solution would be to add scale_by_thickness to this other version (your second option), ideally extracting the shared code to a common subroutine that both of them call: this would avoid making the difference between these two even more confusing than it already is.

Reply:
@billsacks I think I have disentangled this conundrum:
I got all but two 2d vars to go through `restartvar_2d_{TYPE}_bounds`. The two that don't are 2d but only pass the argument dim1name and not dim2name because the second dimension has size 1. So they make use of "optional" for dim2name in `subroutine restartvar_{DIMS}d_{TYPE}`.

So now the easier solution would be to leave `subroutine restartvar_{DIMS}d_{TYPE}` unchanged bc it allows dim1name and dim2name to be optional. The 2d subroutine does not allow that. Pls let me know what you think.

Reply:
Are you thinking that you'll leave your latest changes in place so that the only 2-d variables that go through this {DIMS}d routine are the two degenerate ones in accumulMod?

- [ ] If so, I'd suggest the following, to avoid improper use of this routine in the future:
  - Remove the optional dim2name
  - Assert in that routine that the data's size is 1 along the level dimension
  - Add a comment that this should only be used for 2-d variables in the case that they have a degenerate (size-1) level dimension

Does that seem reasonable?

Reply:
I like this solution.

Reply:
Thanks for this change to the restart routines – I like how you've done this now.